python/ray/data/_internal/execution/streaming_executor.py

```diff
@@ -83,6 +83,9 @@ def __init__(self, options: ExecutionOptions, dataset_tag: str = "unknown_datase
         self._backpressure_policies: Optional[List[BackpressurePolicy]] = None
 
         self._dataset_tag = dataset_tag
+        # Stores if an operator is completed,
+        # used for marking when an op has just completed.
+        self._op_completed: Optional[Dict[PhysicalOperator, bool]] = None
```

Comment:
Nit: Might make it clearer that the value type is `bool`.
```suggestion
        self._has_op_completed: Optional[Dict[PhysicalOperator, bool]] = None
```